File: app/src/main/java/com/example/harsh/Notes/NoteDatabase/NotesDao.java
package com.example.harsh.Notes.NoteDatabase;

import com.example.harsh.Notes.NoteModels.Note;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface NotesDao {
    @Query("SELECT * FROM notes where state = :state ORDER BY updated_date DESC")
    LiveData<List<Note>> loadAllNotes(int state);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long inserEntrie(Note note);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    List<Long> insertNotes(List<Note> notes);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    int updateEntrie(Note note);

    @Delete
    int deleteEntrie(Note note);

    @Query("SELECT * FROM notes WHERE id = :getid")
    Note loadNotesById(int getid);

    @Query("SELECT * FROM notes where state = 0")
    List<Note> getAllData();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insertDeletedNotes(Note note);
}

File: app/src/main/java/com/example/harsh/Notes/NoteDatabase/NotesDao.kt
package com.example.harsh.Notes.NoteDatabase

import androidx.annotation.NonNull
import androidx.lifecycle.LiveData
import androidx.room.*
import com.example.harsh.Notes.NoteDatabase.Tables.Note

@Dao
interface NotesDao {
    @Query("SELECT * FROM notes where state = :state ORDER BY updated_date DESC")
    fun fetchAllNotes(state: Int): LiveData<List<Note>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertNote(note: Note): Long

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertNotes(notes: List<Note>): List<Long>?

    @Update(onConflict = OnConflictStrategy.REPLACE)
    suspend fun updateNote(note: Note): Int

    @Delete
    suspend fun deleteNote(note: Note): Int

    @Query("SELECT * FROM notes WHERE id = :getid")
    suspend fun findNoteById(getid: Int): Note?

    @Query("SELECT * FROM notes where state = 0")
    fun getAllSavedNotes(): List<Note>?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertDeletedNotes(note: Note): Long

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertNotesMain(notes: List<Note>): List<Long>?
}